packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,387 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
```

Comment:
My first thought is that those may be immutables that haven't been set due to them being all 0s. If the constructor doesn't execute and `vm.getDeployedCode` is used then the constructor will not run.

Which one is left and which is right? Is new left?

Reply:
Agreed with mark's assessment. In artifacts from solc, immutables have zeros as placeholders, and they are replaced with real values when the constructor runs. So the left appears to be from the GovernanceToken artifact and the right is probably what's currently deployed

Reply:
Sorry meant to investigate this further to add context. The left is the old genesis output (with all the 0s), and the right is what we're getting now. From what I remember, there are no `immutable`s being set, and there aren't any variables that get set via constructors or upon initialization

Reply:
It is quite possible that the old genesis script had bugs in it. Are you sure that you looked at all the deps? It could be erc721, erc20 (name, symbol)

Reply:
Turns out there are `immutable`s in `EIP712`, the dep chain is: GovernanceToken -> ERC20Votes -> ERC20Permit -> EIP712. Further explained [here](https://github.com/ethereum-optimism/optimism/pull/9292#discussion_r1481951047)